### ISSUE TITLE
 Make enums in typescript be type level strings

### ISSUE THREAD

**Issue by timbod7** [timbod7/adl#114]
Currently "enums" in ADL like:

```
union E
{
   Void v1;
   Void v2;
};
```

map to typescript enums:

```
export enum E {
  v1,
  v2,
}
```

it would be more idiomatic for them to map to type level strings, ie:

```
export type E = 'v1' | 'v2';
```

**Comment by bitc** [comment]
Maybe also emit a value for runtime use? This is often useful in TypeScript

```typescript
export type E = 'v1' | 'v2';
export const E_values: E[] = ['v1', 'v2'];
```

**Comment by timbod7** [comment]
resolved with ec58a1c
